lib/tesla/middleware/message_pack.ex

```diff
@@ -0,0 +1,159 @@
+defmodule Tesla.Middleware.MessagePack do
+  @moduledoc """
+  Encode requests and decode responses as MessagePack.
+
+  This middleware requires [Msgpax](https://hex.pm/packages/msgpax) as dependency.
+
+  Remember to add `{:msgpax, ">= 2.3.0"}` to dependencies.
+  Also, you need to recompile Tesla after adding `:msgpax` dependency:
+
+  ```
+  mix deps.clean tesla
+  mix deps.compile tesla
+  ```
+
+  ## Examples
+
+  ```
+  defmodule MyClient do
+    use Tesla
+
+    plug Tesla.Middleware.MessagePack
+    # or
+    plug Tesla.Middleware.MessagePack, engine_opts: [binary: true]
+    # or
+    plug Tesla.Middleware.MessagePack, decode: &Custom.decode/1, encode: &Custom.encode/1
+  end
+  ```
+
+  ## Options
+
+  - `:decode` - decoding function
+  - `:encode` - encoding function
+  - `:encode_content_type` - content-type to be used in request header
+  - `:decode_content_types` - list of additional decodable content-types
+  - `:engine_opts` - optional engine options
+  """
+
+  @behaviour Tesla.Middleware
+
+  @content_type "application/msgpack"
```

Comment:
Hey @chaodhib, maybe instead of reusing the same default attribute for encode/decode, what about having a list for decodable ones (similar to JSON middleware) with both `application/msgpack` and `application/x-msgpack`?

The rationale to support both out of box (sans overrides) is the first is what often we find on the wild but the [second arguably valid as well][1].

What do you think?

[1]: https://github.com/msgpack/msgpack/issues/194

Reply:
Sure, I can add that

Reply:
Done!